File: src/main/java/com/libreria/sistema/service/LibroService.java
package com.libreria.sistema.service;

import com.libreria.sistema.dto.request.LibroRequest;
import com.libreria.sistema.dto.response.LibroResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface LibroService {

    Page<LibroResponse> listar(String busqueda, Long autorId, Long categoriaId, Pageable pageable);

    LibroResponse buscarPorId(Long id);

    LibroResponse buscarPorIsbn(String isbn);

    LibroResponse crear(LibroRequest request);

    LibroResponse actualizar(Long id, LibroRequest request);

    void eliminar(Long id);

    List<LibroResponse> librosBajoStock(int limite);
}

File: src/main/resources/static/js/services/libroService.js
import { api } from '../utils.js';

const LibroService = {
  listar: (params) => api(`/api/libros?${new URLSearchParams(params)}`),
  buscarPorId: (id) => api(`/api/libros/${id}`),
  buscarPorIsbn: (isbn) => api(`/api/libros/isbn/${isbn}`),
  crear: (data) => api('/api/libros', { method: 'POST', body: JSON.stringify(data) }),
  actualizar: (id, data) => api(`/api/libros/${id}`, { method: 'PUT', body: JSON.stringify(data) }),
  eliminar: (id) => api(`/api/libros/${id}`, { method: 'DELETE' }),
};

export default LibroService;
